Source/com/drew/imaging/png/PngMetadataReader.java

```diff
@@ -178,30 +183,69 @@ private static void processChunk(@NotNull Metadata metadata, @NotNull PngChunk c
             metadata.addDirectory(directory);
         } else if (chunkType.equals(PngChunkType.tEXt)) {
             SequentialReader reader = new SequentialByteArrayReader(bytes);
-            byte[] keywordBytes = reader.getNullTerminatedBytes(79);
-            StringValue keyword = new StringValue(keywordBytes, _latin1Encoding);
-            int bytesLeft = bytes.length - keywordBytes.length - 1;
+            String keyword = reader.getNullTerminatedStringValue(79, _latin1Encoding).toString();
+            int bytesLeft = bytes.length - keyword.length() - 1;
```

Comment:
Generally the code avoids treating string lengths as numbers of bytes in arithmetic, to avoid bugs with multi-byte encodings. However that's not a problem with LATIN-1.